x-pack/plugin/blob-cache/src/test/java/org/elasticsearch/blobcache/shared/SharedBlobCacheServiceTests.java

```diff
@@ -498,7 +497,7 @@ public void testGetMultiThreaded() throws IOException {
                                     if (yield[i] == 0) {
                                         Thread.yield();
                                     }
-                                    assertNotNull(cacheFileRegion.testOnlyNonVolatileIO());
+                                    assertTrue(cacheFileRegion.testOnlyNonVolatileIO() != null || cacheFileRegion.isEvicted());
```

Comment:
I think we can now also do this outside the incref block, I think it could be good to move it there. I originally intended this to be a safe spot for it, but given the extra check for evicted we may as well just do it prior to the incref block.

Reply:
Fair point. Pushed 7f50458a60ee5e48028bdb6d4bfd331ee708803a